source/guides/put-it-together/ddiscordbot.md

```diff
@@ -0,0 +1,303 @@
+
+The main uses might be an account linker, which verifies a Discord user with their player on the server, and a chat bridge between Discord and Minecraft. You might also relay announcements, host cross-platform events, and let Discord users query data from the server.
+
+### Creating a Bot
+
+There are enough tutorials out there for creating a bot account. [Here's a handy one](https://discordpy.readthedocs.io/en/stable/discord.html).
+
+### Logging In
+
+To log in using dDiscordBot, use the `discordconnect` command. **Don't hardcode your token into the script!** Use a separate tokenfile placed outside of the `scripts` folder. A common location for this is in `Denizen/data`.
+
+The `discordconnect` command takes an `id` argument. This can be anything, and it identifies the connection to the bot user. Almost all Discord commands take this argument, and it must be the same for a specified connection.
+
+Note that you only need to connect once after the server starts. Additionally, make sure to ~wait this command, as you should with all other Discord commands.
+
+```dscript_green
+connect_to_discord:
+    type: world
+    events:
+        after server start:
+        - ~discordconnect id:mybot tokenfile:data/tokenfile.txt
+```
+
+### Sending a Message
+
+Now that you've logged in, you can send your first message. The `discordmessage` command has a `channel` argument, which is the Discord channel you want the bot to send the message in. It must be visible to the bot. You can use the channel's ID; if you don't know how to find it, [look here](https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-).
+
+Note that if you're going to use a channel or server's ID a lot, you should put it in a data script/flag it somewhere.
+
+Here's how it would look in task form:
+
+```dscript_blue
+send_a_message:
+    type: task
+    script:
+    - ~discordmessage id:mybot channel:(channel_id) "Hello, world!"
+```
+
+### Automatic Messaging
+
+Let's make this automated! Using the `discord message received` event, you can detect when a Discord user sends a message and run commands based off the message content. You can also respond to the same channel they sent the message in! For example, if you wanted to respond to the user if their message includes "ping", you'd do:
+
+```dscript_green
+ping_pong:
+    type: world
+    events:
+        on discord message received:
+        - if <context.new_message.text.contains_text[ping]>:
+            - ~discordmessage id:mybot channel:<context.new_message.channel> Pong!
+```
+
+![](https://media.discordapp.net/attachments/533123622093455362/894456615514808341/unknown.png)
+
+### Chat Bridge
+
+Now you've got the basic tools to make a chat bridge. In order for this to function, it needs to:
+
+- send a message to a Discord channel when a Minecraft user chats
+- narrate to the online Minecraft players when a Discord user sends a message in the channel
```

Comment:
Your previous bulleted list was capitalized, this change would keep your formatting consistent
```suggestion
- Send a message to a Discord channel when a Minecraft user chats
- Narrate to the online Minecraft players when a Discord user sends a message in the channel
```